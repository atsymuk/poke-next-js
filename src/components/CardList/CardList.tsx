import { VFC } from 'react'

import Card from 'components/Card/Card'

import styles from './cardList.module.scss'

export interface CardItem {
  image: string
  title: string
  url: string
}

interface CardListProps {
  items: CardItem[]
}

const CardList: VFC<CardListProps> = ({ items }) => (
  <ul className={styles.container}>
    {items.map(({ image, title, url }) => (
      <li className={styles.item} key={title}>
        <Card image={image} title={title} url={url} />
      </li>
    ))}
  </ul>
)

export default CardList
