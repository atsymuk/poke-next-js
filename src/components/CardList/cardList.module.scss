@import 'styles/breakpoints.scss';

.container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  list-style: none;
  margin: 0;
  padding: 0;
}

.item {
  flex-shrink: 0;
  margin: 0 20px 20px 0;

  &:nth-child(3n + 3) {
    margin-right: 0;
  }
}

@media (max-width: $md) {
  .item {
    &:nth-child(3n + 3) {
      margin-right: 20px;
    }

    &:nth-child(2n + 2) {
      margin-right: 0px;
    }
  }
}

@media (max-width: $sm) {
  .item {
    &:nth-child(3n + 3) {
      margin-right: 0px;
    }

    &:nth-child(2n + 2) {
      margin-right: 0px;
    }
  }
}
