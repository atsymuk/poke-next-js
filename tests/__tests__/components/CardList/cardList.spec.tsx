import { render } from '@testing-library/react'
import CardList from 'components/CardList/CardList'

it('CardList component snapshot', () => {
  const items = [
    {
      image: 'http://test.test/test.png',
      title: 'test1',
      url: '/',
    },
    {
      image: 'http://test.test/test.png',
      title: 'test2',
      url: '/',
    },
    {
      image: 'http://test.test/test.png',
      title: 'test3',
      url: '/',
    },
  ]

  const { container } = render(<CardList items={items} />)
  expect(container).toMatchSnapshot()
})
