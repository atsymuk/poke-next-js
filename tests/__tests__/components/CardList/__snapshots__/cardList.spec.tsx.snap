// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList component snapshot 1`] = `
<div>
  <ul
    class="container"
  >
    <li
      class="item"
    >
      <a
        href="/"
      >
        <div
          class="container"
        >
          <div
            class="imageContainer"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27300%27%20height=%27300%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="test1"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              test1
            </h2>
          </div>
        </div>
      </a>
    </li>
    <li
      class="item"
    >
      <a
        href="/"
      >
        <div
          class="container"
        >
          <div
            class="imageContainer"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27300%27%20height=%27300%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="test2"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              test2
            </h2>
          </div>
        </div>
      </a>
    </li>
    <li
      class="item"
    >
      <a
        href="/"
      >
        <div
          class="container"
        >
          <div
            class="imageContainer"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27300%27%20height=%27300%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="test3"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              test3
            </h2>
          </div>
        </div>
      </a>
    </li>
  </ul>
</div>
`;
